### DISCUSSION TITLE
Why not use fsync to ensure log file is written to disk?

### DISCUSSION THREAD

**Discussion by ZuchaoWang** [cmu-db/bustub#516] [General]
I see in disk_manager.cpp, at the end of WriteLog and WritePage functions, we call flush(). The comment says "needs to flush to keep disk file in sync". But doesn't fstream's flush() just flush data from standard buffer to OS buffer? I don't think it will flush related OS buffer pages to disk. Why not call fsync() to force everything to disk?

**Comment by skyzh** [comment]
The logging thing is broken and I don't think it's maintained. We would probably make it "correct" next time we want to do logging and recovery.
